app/client-v2/src/components/HomeView/CreateProjectButton.vue

```diff
@@ -0,0 +1,92 @@
+<script setup lang="ts">
+import { useToastService } from "@/composables/useToastService";
+import { getApiUrl } from "@/config";
+import { SPECIES, type ProjectOverview } from "@/types/projectTypes";
+import { useFetch } from "@vueuse/core";
+import { reactive, ref } from "vue";
+import { useRouter } from "vue-router";
+
+const props = defineProps<{
+  projects: ProjectOverview[];
+}>();
+
+const { showErrorToast } = useToastService();
+const router = useRouter();
+const apiUrl = getApiUrl();
+const visible = ref(false);
+const species = ref(null);
+const name = ref("");
+const errors = reactive({
+  name: "",
+  species: "",
+  creation: ""
+});
+
+const isDuplicateName = (name: string) => props.projects.some((project) => project.name === name);
+
+const closeDialog = () => {
+  visible.value = false;
+  species.value = null;
+  errors.name = errors.species = errors.creation = name.value = "";
```

Comment:
This is a little bit gruesome. Could use this pattern of having an initial error state which is used for both initialisation and reset on dialog close: https://stackoverflow.com/questions/61184749/how-to-properly-reset-vue-composition-apis-reactive-values